go/obscuronode/host/host.go

```diff
@@ -198,16 +198,17 @@ func (a *Node) Stop() {
 	atomic.StoreInt32(a.stopNodeInterrupt, 1)
 
 	a.P2p.StopListening()
-	if a.clientServer != nil {
-		a.clientServer.Stop()
-	}
 
 	if err := a.EnclaveClient.Stop(); err != nil {
 		nodecommon.LogWithID(a.shortID, "Could not stop enclave server. Cause: %v", err.Error())
 	}
 	time.Sleep(time.Second)
 	a.exitNodeCh <- true
 	a.EnclaveClient.StopClient()
```

Comment:
Maybe not as part of this PR but this Stop() method feels like it could be a bit more robust with careful defers. If P2p.StopListening() panicked, for example, then EnclaveClient and ClientServer might be left running.